Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/approximate_convex_decomposition.h

```diff
@@ -0,0 +1,1689 @@
+    (std::min<FT>)(a.zmin(), b.zmin()),
+    (std::max<FT>)(a.xmax(), b.xmax()),
+    (std::max<FT>)(a.ymax(), b.ymax()),
+    (std::max<FT>)(a.zmax(), b.zmax()));
+}
+
+template<typename FT>
+std::tuple<Vec3_uint, FT> calculate_grid_size(Bbox_3& bb, const std::size_t number_of_voxels) {
+  std::size_t max_voxels_axis = std::pow(number_of_voxels, 1.0 / 3.0);
+  assert(max_voxels_axis > 3);
+  // get longest axis
+  FT longest = 0;
+
+  if (bb.x_span() >= bb.y_span() && bb.x_span() >= bb.z_span())
+    longest = bb.x_span();
+  else if (bb.y_span() >= bb.x_span() && bb.y_span() >= bb.z_span())
+    longest = bb.y_span();
+  else if (bb.z_span() >= bb.x_span() && bb.z_span() >= bb.y_span())
+    longest = bb.z_span();
+
+  const FT voxel_size = longest * FT(1.0 / (max_voxels_axis - 3));
+
+  FT s = 1.5 * voxel_size;
+
+  bb = Bbox_3(to_double(bb.xmin() - s), to_double(bb.ymin() - s), to_double(bb.zmin() - s), to_double(bb.xmax() + s), to_double(bb.ymax() + s), to_double(bb.zmax() + s));
+
+  return { Vec3_uint{static_cast<unsigned int>(to_double(bb.x_span() / voxel_size + 0.5)), static_cast<unsigned int>(to_double(bb.y_span() / voxel_size + 0.5)), static_cast<unsigned int>(to_double(bb.z_span() / voxel_size + 0.5))}, voxel_size};
+}
+
+template<typename GeomTraits, typename PolygonMesh, typename NamedParameters = parameters::Default_named_parameters>
+const typename GeomTraits::Point_3 &point(typename boost::graph_traits<PolygonMesh>::face_descriptor fd,
+  const PolygonMesh& pmesh,
+  const NamedParameters& np = parameters::default_values())
+{
+  using parameters::choose_parameter;
+  using parameters::get_parameter;
+  typename GetVertexPointMap<PolygonMesh, NamedParameters>::const_type
+    vpm = choose_parameter(get_parameter(np, internal_np::vertex_point),
+      get_const_property_map(CGAL::vertex_point, pmesh));
+
+  return get(vpm, target(halfedge(fd, pmesh), pmesh));
+}
+
+template<typename FaceGraph, typename FT>
+Bbox_uint grid_bbox_face(const FaceGraph& mesh, const typename boost::graph_traits<FaceGraph>::face_descriptor fd, const Bbox_3& bb, const FT& voxel_size) {
+  Bbox_3 face_bb = face_bbox(fd, mesh);
+  return Bbox_uint({
+    static_cast<unsigned int>((face_bb.xmin() - bb.xmin()) / to_double(voxel_size) - 0.5),
+    static_cast<unsigned int>((face_bb.ymin() - bb.ymin()) / to_double(voxel_size) - 0.5),
+    static_cast<unsigned int>((face_bb.zmin() - bb.zmin()) / to_double(voxel_size) - 0.5)
+    }, {
+    static_cast<unsigned int>((face_bb.xmax() - bb.xmin()) / to_double(voxel_size) + 0.5),
+    static_cast<unsigned int>((face_bb.ymax() - bb.ymin()) / to_double(voxel_size) + 0.5),
+    static_cast<unsigned int>((face_bb.zmax() - bb.zmin()) / to_double(voxel_size) + 0.5)
+    });
+}
+
+template<typename GeomTraits>
+Iso_cuboid_3<GeomTraits> bbox_voxel(const Vec3_uint& voxel, const Bbox_3& bb, const typename GeomTraits::FT& voxel_size) {
+  return Bbox_3(
+    bb.xmin() + voxel[0] * to_double(voxel_size),
```

Comment:
Wouldn't it make sense to call `to_double()` only once, even outside of this function?

Reply:
Done, but inside of the function.